_posts/2019/2019-02-10-be-careful-assigning-to-has-one-relations.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Be Careful Assigning to has_one Relations
+description: Strange Think
+layout: article
+category: ruby
+image:
+  base: '2019/be-careful-assigning-to-has-one-relations'
+  alt: 'Broken net'
+  source: "https://unsplash.com/photos/IV--3UEiHlI"
+  credit: "Ben Hershey"
+---
+
+For much of the time, when building relationships between models you typically use `has_many` and `belongs_to`. There are some circumstances where a `has_one` relationship is more appropriate.
+
+However, the behaviour of `has_one` has some wrinkles in its behaviour that make it a little trickier to deal with. See the side-effects of the generated `#association=` method [in the Rails documentation](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_one).
+
+
+## When…
+
+…assigning an invalid new object means you lose the relationship with the currently related model.
```

Comment:
Huh? This is a confusing sentence and do you mean a colon instead of a full-stop? 

It's confusing because this concept of an 'invalid new object' comes out of nowhere. I'm having to stop to think about what it means and what 'losing the relationship...' means as well. 

Maybe you need to say more in the intro about this problem that you're going to introduce.